diesel/src/expression/array_comparison.rs

```diff
@@ -193,19 +181,16 @@ impl<T, DB> QueryFragment<DB> for Many<T> where
     DB: Backend,
     T: QueryFragment<DB>,
 {
-    fn to_sql(&self, out: &mut DB::QueryBuilder) -> BuildQueryResult {
-        try!(self.0[0].to_sql(out));
-        for value in self.0[1..].iter() {
-            out.push_sql(", ");
-            try!(value.to_sql(out));
-        }
-        Ok(())
-    }
-
-    fn walk_ast(&self, mut pass: AstPass<DB>) -> QueryResult<()> {
-        pass.unsafe_to_cache_prepared();
+    fn walk_ast(&self, mut out: AstPass<DB>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+        let mut first = true;
         for value in &self.0 {
-            value.walk_ast(pass.reborrow())?;
+            if first {
+                first = false;
+            } else {
+                out.push_sql(", ");
+            }
+            value.walk_ast(out.reborrow())?;
```

Comment:
Nice win for readability! I'd probably add a unit test for that, but it's simple enough that it's not really necessary (and it's tested in a whole bunch of other tests anyway)